test/runtime/test_format_ignores.py

```diff
@@ -0,0 +1,220 @@
+import errno
+import sys
+import tempfile
+from collections.abc import Generator
+from pathlib import Path
+
+import pytest
+
+tool_path = str(Path(__file__).parent.parent.parent)
+if tool_path not in sys.path:
+    sys.path.insert(0, tool_path)
+
+from tool.format_ignores import (  # noqa: E402
+    _process_directory,  # noqa: PLC2701
+    _process_file,  # noqa: PLC2701
+    _sort_ignore_list,  # noqa: PLC2701
+    main,
+)
+
+
+@pytest.fixture
+def temp_dir() -> Generator[Path, None, None]:
+    """Temporary directory for testing
+
+    Yields:
+        Path: The path to the temporary directory.
+    """
+    with tempfile.TemporaryDirectory() as temp_dir_path:
+        yield Path(temp_dir_path)
+
+
+@pytest.mark.parametrize(
+    ("input_text", "expected_output"),
+    [
+        ("error1, error2", "error1, error2"),
+        ("error2, error1", "error1, error2"),
+        ("error1,error2", "error1, error2"),
+        ("error1,  error2", "error1, error2"),
+        ("error1, error1, error2", "error1, error2"),
+        ("", ""),
+        (" , error1, ,error2, ", "error1, error2"),
+    ],
+)
+def test_sort_ignore_list(input_text: str, expected_output: str) -> None:
+    """Test _sort_ignore_list function"""
+    result = _sort_ignore_list(input_text)
+    assert result == expected_output
+
+
+def test_process_file(temp_dir: Path) -> None:
+    """Test _process_file function"""
+    test_file = temp_dir / "test.pyi"
+    test_content = """def func1() -> None:
+    # type: ignore[attr-defined, no-any-return]
+    pass
+
+def func2() -> None:
+    # pyright: ignore[reportUnknownMemberType,reportUnknownVariableType]
+    pass
+"""
```

Comment:
Indentation-wise, it could help to extract these as contants and place them outside of this test.

An alternative is to ignore the additional indentation here, and have `textwrap.dedent` correct it: https://docs.python.org/3/library/textwrap.html#textwrap.dedent